tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright 2016 Kinvolk GmbH
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Author: Krzesimir Nowak <krzesimir@kinvolk.io>
+ */
+
+#include "eos-test-utils.h"
+#include "misc-utils.h"
+#include "ostree-spawn.h"
+
+#include <string.h>
+
+#ifndef EOS_UPDATER_BINARY
+#error "EOS_UPDATER_BINARY is not defined"
+#endif
+
+#ifndef EOS_UPDATE_SERVER_BINARY
+#error "EOS_UPDATE_SERVER_BINARY is not defined"
+#endif
+
+#ifndef EOS_AUTOUPDATER_BINARY
+#error "EOS_AUTOUPDATER_BINARY is not defined"
+#endif
+
+#ifndef GPG_HOME_DIRECTORY
+#error "GPG_HOME_DIRECTORY is not defined"
+#endif
+
+#ifndef GPG_BINARY
+#error "GPG_BINARY is not defined"
+#endif
+
+void
+eos_updater_fixture_setup (EosUpdaterFixture *fixture,
+                           gconstpointer user_data)
+{
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *tmpdir_path = NULL;
+
+  fixture->dbus = g_test_dbus_new (G_TEST_DBUS_NONE);
+  g_test_dbus_add_service_dir (fixture->dbus, TEST_SERVICES_DIRECTORY);
+  g_test_dbus_up (fixture->dbus);
+
+  tmpdir_path = g_dir_make_tmp ("eos-updater-test-XXXXXX", &error);
+  g_assert_no_error (error);
+  fixture->tmpdir = g_file_new_for_path (tmpdir_path);
+}
+
+void
+eos_updater_fixture_teardown (EosUpdaterFixture *fixture,
+                              gconstpointer user_data)
+{
+  g_autoptr (GError) error = NULL;
+
+  rm_rf (fixture->tmpdir, &error);
+  g_assert_no_error (error);
+  g_object_unref (fixture->tmpdir);
+
+  g_test_dbus_down (fixture->dbus);
+  g_object_unref (fixture->dbus);
+}
+
+gchar *
+get_keyid (void)
+{
+  g_autoptr(GFile) gpg_home = g_file_new_for_path (GPG_HOME_DIRECTORY);
+  g_autoptr(GFile) keyid = g_file_get_child (gpg_home, "keyid");
+  g_autoptr(GBytes) bytes = NULL;
+  g_autoptr(GError) error = NULL;
+  gsize len;
+
+  load_to_bytes (keyid, &bytes, &error);
+  g_assert_no_error (error);
+  len = g_bytes_get_size (bytes);
+  g_assert_cmpint (len, ==, 8);
```

Comment:
this should probably be a plain assert instead of using `g_assert_cmpint`, since it will be listed as a cause of failure in tests where that isn't what the test is actually testing.